### ISSUE TITLE
Problem using ECP

### ISSUE THREAD

**Issue by MRJCrunch** [miracl/amcl#8]
I have this formula:
![formula](https://user-images.githubusercontent.com/17198997/29071437-4c803024-7c4c-11e7-924b-1c92f7ef0c04.png)

All elements in brackets are ECP. x and c are BIG. In a simplified form, I need to calculate ECP^(1/BIG). How can I do it?

**Comment by mcarrickscott** [comment]
Hello Evgeniy,


I am not familiar with this formula.

In ECC there are two notations commonly used, the additive notation and the
multiplicative notation.

In the additive notation a point multiplication is denoted P=x.G

In multiplicative notation it is denoted P=G^x

Its just a notational difference - they mean the same thing.

The additive notation is the most common, but the multiplicative notation
has recently become quite popular, particularly in the context of bilinear
pairings.

The AMCL library assumes the (still more common) additive notation, so you
need to "translate" the formula to this form, in which case it becomes

1/(x+c) .((H0+m2.H1 )+U_r+G1+s''.H2)

where I have capitalised points for clarity.

So basically . becomes +, and ^ becomes .


Mike

**Comment by MRJCrunch** [comment]
I understand it, but how can I calculate 1/(x+c)?  x and c are BIG. Could you give me a code example(rust version if it's possible)?
let a = ECP::new()
let b = BIG::new() 
a.mul(1/b). How will I calculate 1/b?

**Comment by mcarrickscott** [comment]
The appropriate function is invmodp(), from big.rs

So x.invmodp(&r), calculates 1/x mod r, where x and r are BIGs.

See ecdh.rs line 438 for an example of use.

Note that 1/(x+c)  is calculated modulo the group order, so access it as

let r=BIG::new_ints(&rom::CURVE_ORDER);

In general f(x).G should be calculated as (f(x) mod r).G



Mike

**Comment by MRJCrunch** [comment]
Thank you, Mike!
